src/ring_theory/class_group.lean

```diff
@@ -0,0 +1,233 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+
+import group_theory.quotient_group
+import ring_theory.dedekind_domain
+
+/-!
+# The ideal class group
+
+This file defines the ideal class group `class_group R K` of fractional ideals of `R`
+inside `A`'s field of fractions `K`.
+
+## Main definitions
+ - `to_principal_ideal` sends an invertible `x : K` to an invertible fractional ideal
+ - `class_group` is the quotient of invertible fractional ideals modulo `to_principal_ideal.range`
+ - `class_group.mk0` sends a nonzero integral ideal in a Dedekind domain to its class
+
+## Main results
+ - `class_group.mk0_eq_mk0_iff` shows the equivalence with the "classical" definition,
+   where `I ~ J` iff `x I = y J` for `x y ≠ (0 : R)`
+-/
+
+section integral_domain
+
+variables {R K L : Type*} [integral_domain R]
+variables [field K] [field L] [decidable_eq L]
+variables [algebra R K] [is_fraction_ring R K]
+variables [algebra K L] [finite_dimensional K L]
+variables [algebra R L] [is_scalar_tower R K L]
+
+local notation R`⁰`:9000 := non_zero_divisors R
```

Comment:
Should this notation be put in some locale? It's used in several files by now, isn't it?